oshmem/mca/scoll/basic/scoll_basic_alltoall.c

```diff
@@ -135,9 +135,10 @@ static int a2as_alg_simple(struct oshmem_group_t *group,
 
     dst_blk_idx = oshmem_proc_group_find_id(group, group->my_pe);
 
-    for (src_blk_idx = 0; src_blk_idx < group->proc_count; src_blk_idx++) {
+    for (i = 0; i < group->proc_count; i++) {
 
-        dst_pe = get_dst_pe(group, src_blk_idx, dst_blk_idx);
+        dst_pe = get_dst_pe(group, i, dst_blk_idx);
+        src_blk_idx = oshmem_proc_group_find_id(group, dst_pe);
```

Comment:
Calling oshmem_proc_group_find_id should avoided as it does a linear lookup.
the index should be  (dst_blk_idx + src_blk_idx) % group->proc_count and it is already computed in get_dst_pe()